src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+
 public class SalaryInfo {
+    private static final int INDEX_OF_DATE = 0;
+    private static final int INDEX_OF_NAME = 1;
+    private static final int INDEX_OF_HOURS = 2;
+    private static final int INDEX_OF_RATE = 3;
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] salaries = new int[names.length];
+        List<String> namesList = Arrays.asList(names);
+
+        LocalDate parsedDateFrom = parseDate(dateFrom);
+        LocalDate parsedDateTo = parseDate(dateTo);
+
+        for (String record : data) {
+            String[] parsedRecord = record.split(" ");
+            LocalDate date = parseDate(parsedRecord[INDEX_OF_DATE]);
+            String name = parsedRecord[INDEX_OF_NAME];
+            int hours = Integer.parseInt(parsedRecord[INDEX_OF_HOURS]);
+            int rate = Integer.parseInt(parsedRecord[INDEX_OF_RATE]);
+
+            int indexOfRow = namesList.indexOf(name);
+
+            if (date.compareTo(parsedDateFrom) >= 0
```

Comment:
Move this boolean expression to method, also there are more suitable methods like isBefore/isAfter to make date comparation